### PR TITLE
Simplify examples/ws2812

### PR DESCRIPTION
**Pull request by sago35**
The WS2812 example had variations due to its dependency on led and the different PIN names used.
With [this PR](https://github.com/tinygo-org/tinygo/pull/4231), boards with WS2812 now have `machine.WS2812` available.
This PR makes the following changes:

* Remove dependency on `machine.LED` (as some variations don't have `machine.LED`)
* Update to use `machine.WS2812` where it is defined

### CODE REVIEW

**Comment by sago35** [comment]
My main motivation was to create an example of running WS2812 on waveshare-rp2040-zero. However, adding the new variation required many changes, so I created this PR. Going forward, this example will work without changes as long as machine.WS2812 is defined.

I have tested the following for verification and confirmed that at least the build is successful.

```
tinygo build -o /tmp/out.uf2 -target arduino               ./examples/ws2812/
tinygo build -o /tmp/out.uf2 -target digispark             ./examples/ws2812/
tinygo build -o /tmp/out.uf2 -target qtpy                  ./examples/ws2812/
tinygo build -o /tmp/out.bin -target m5stamp-c3            ./examples/ws2812/
tinygo build -o /tmp/out.uf2 -target thingplus-rp2040      ./examples/ws2812/
tinygo build -o /tmp/out.uf2 -target waveshare-rp2040-zero ./examples/ws2812/
```

File: examples/ws2812/main.go [hunk 1]
```diff
@@ -12,11 +12,12 @@ import (
 	"tinygo.org/x/drivers/ws2812"
 )
 
-var leds [10]color.RGBA
+var (
+	noe  machine.Pin
```
Comment (aykevl) [inline examples/ws2812/main.go]: This looks like a typo.
Did you test this code?

**Review by aykevl — COMMENTED** [review]
Seems like a good idea to me!

File: examples/ws2812/main.go (hunk 1 above)
Comment (sago35) [reply examples/ws2812/main.go]: There was a typo. Also, the variable "noe" wasn't used at all. I've fixed it to make it more understandable. @aykevl

**Comment by deadprogram** [comment]
Thank you for the improvements @sago35 and to @aykevl for review. Now merging.

**Pull request merged**